src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,160 @@
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Lowering.LocalRewriter
+{
+    internal class LocalUsingVarRewriter : BoundTreeRewriterWithStackGuard
+    {
+        public static BoundNode Rewrite(BoundStatement statement)
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
+        {
+            ImmutableArray<BoundStatement> statements = (ImmutableArray<BoundStatement>)this.VisitList(node.Statements);
+
+            for (int i = 0; i < statements.Length; i++)
+            {
+                if (statements[i] is BoundLocalDeclaration localDeclaration)
+                {
+                    if (localDeclaration.LocalSymbol.IsUsing)
+                    {
+                        ImmutableArray<LocalSymbol> locals = ImmutableArray.Create<LocalSymbol>(localDeclaration.LocalSymbol);
+
+                        List<BoundStatement> precedingStatements = new List<BoundStatement>();
+                        for (int j = 0; j < i; j++)
+                        {
+                            precedingStatements.Add(statements[j]);
+                        }
+                        var followingStatements = ImmutableArray.Create(
+                            statements, i + 1,
+                            statements.Length - i - 1);
+
+                        var localDeclarations = ImmutableArray.Create(localDeclaration);
+
+                        BoundBlock innerBlock = new BoundBlock(
+                            syntax: localDeclaration.Syntax,
+                            locals: ImmutableArray.Create<LocalSymbol>(),
+                            statements: followingStatements
+                            );
+
+                        BoundUsingStatement boundUsing = new BoundUsingStatement(
+                            syntax: localDeclaration.Syntax,
+                            locals: ImmutableArray.Create<LocalSymbol>(),
+                            declarationsOpt: new BoundMultipleLocalDeclarations(
+                                localDeclaration.Syntax,
+                                localDeclarations),
+                            expressionOpt: null,
+                            iDisposableConversion: Conversion.Identity,
+                            disposeMethodOpt: null,
+                            body: innerBlock
+                            );
+                        precedingStatements.Add(boundUsing);
+
+                        BoundBlock outermostBlock = new BoundBlock(
+                            syntax: localDeclaration.Syntax,
+                            locals: node.Locals,
+                            statements: precedingStatements.ToImmutableArray<BoundStatement>());
+
+                        return outermostBlock;
```

Comment:
If we return early won't we skip all subsequent `using` variables in this block?